cranelift/wasm/src/code_translator/bounds_checks.rs

```diff
@@ -364,19 +374,63 @@ fn compute_addr(
     addr_ty: ir::Type,
     index: ir::Value,
     offset: u32,
+    pcc: bool,
 ) -> ir::Value {
     debug_assert_eq!(pos.func.dfg.value_type(index), addr_ty);
 
     let heap_base = pos.ins().global_value(addr_ty, heap.base);
+
+    let pcc_memtype = if pcc {
+        Some(
+            heap.memory_type
+                .expect("A memory type is required when PCC is enabled"),
+        )
+    } else {
+        None
+    };
+
+    if let Some(ty) = pcc_memtype {
+        pos.func.dfg.facts[heap_base] = Some(Fact::Mem {
+            ty,
+            min_offset: 0,
+            max_offset: 0,
+        });
+    }
+
     let base_and_index = pos.ins().iadd(heap_base, index);
+
+    if let Some(ty) = pcc_memtype {
+        // TODO: handle memory64 as well (pass in the CLIF type of
+        // `index`).
+        pos.func.dfg.facts[base_and_index] = Some(Fact::Mem {
+            ty,
+            min_offset: 0,
+            max_offset: u64::from(u32::MAX),
+        });
```

Comment:
We can actually use `heap.index_type` already, we don't need another parameter. I think we should at least assert on that value now, if we don't update the actual max now, so that it isn't silently incorrect.

Reply:
Ah, yes, good point -- added that assert.